http/address_watcher.go

```diff
@@ -0,0 +1,118 @@
+package http
+
+import (
+	"encoding/hex"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/rsksmart/liquidity-provider-server/connectors"
+	"github.com/rsksmart/liquidity-provider/providers"
+	"github.com/rsksmart/liquidity-provider/types"
+	log "github.com/sirupsen/logrus"
+	"math/big"
+)
+
+type BTCAddressWatcher struct {
+	btc             connectors.BTCInterface
+	rsk             connectors.RSKInterface
+	lp              providers.LiquidityProvider
+	calledForUser   bool
+	registeredPegIn bool
+	quote           *types.Quote
+}
+
+func NewBTCAddressWatcher(btc connectors.BTCInterface, rsk connectors.RSKInterface, provider providers.LiquidityProvider, q *types.Quote) (*BTCAddressWatcher, error) {
+	watcher := BTCAddressWatcher{
+		btc:             btc,
+		rsk:             rsk,
+		lp:              provider,
+		quote:           q,
+		calledForUser:   false,
+		registeredPegIn: false,
+	}
+	return &watcher, nil
+}
+
+func (w *BTCAddressWatcher) RegisteredPegIn() bool {
+	return w.registeredPegIn
+}
+
+func (w *BTCAddressWatcher) OnNewConfirmation(txHash string, confirmations int64, amount float64) {
+	if !w.calledForUser && confirmations >= int64(w.quote.Confirmations) {
+		_, err := w.performCallForUser()
```

Comment:
Why return two objects if you are going to ignore it?

Reply:
done